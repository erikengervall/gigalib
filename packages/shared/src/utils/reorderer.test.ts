import { reorderer } from './index';

describe('reorderer', () => {
  it('should be able to move items down the list', () => {
    const { reorderedList } = reorderer({
      sortedList: [
        { initialListIndex: 0, listIndex: 0 },
        { initialListIndex: 1, listIndex: 1 },
        { initialListIndex: 2, listIndex: 2 },
        { initialListIndex: 3, listIndex: 3 },
      ],
      fromListIndex: 0,
      toListIndex: 2,
    });

    expect(reorderedList).toMatchSnapshot();
  });

  it('should be able to move items up the list', () => {
    const { reorderedList } = reorderer({
      sortedList: [
        { initialListIndex: 0, listIndex: 0 },
        { initialListIndex: 1, listIndex: 1 },
        { initialListIndex: 2, listIndex: 2 },
        { initialListIndex: 3, listIndex: 3 },
      ],
      fromListIndex: 3,
      toListIndex: 1,
    });

    expect(reorderedList).toMatchSnapshot();
  });
});
