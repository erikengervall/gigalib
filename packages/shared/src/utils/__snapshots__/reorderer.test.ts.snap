// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reorderer should be able to move items down the list 1`] = `
[
  {
    "initialListIndex": 1,
    "listIndex": 0,
  },
  {
    "initialListIndex": 2,
    "listIndex": 1,
  },
  {
    "initialListIndex": 0,
    "listIndex": 2,
  },
  {
    "initialListIndex": 3,
    "listIndex": 3,
  },
]
`;

exports[`reorderer should be able to move items up the list 1`] = `
[
  {
    "initialListIndex": 0,
    "listIndex": 0,
  },
  {
    "initialListIndex": 3,
    "listIndex": 1,
  },
  {
    "initialListIndex": 1,
    "listIndex": 2,
  },
  {
    "initialListIndex": 2,
    "listIndex": 3,
  },
]
`;
